velox/functions/sparksql/DateTimeFunctions.h

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <velox/type/Timestamp.h>
+#include "velox/core/QueryConfig.h"
+#include "velox/external/date/tz.h"
+#include "velox/functions/lib/DateTimeFormatter.h"
+#include "velox/functions/prestosql/types/TimestampWithTimeZoneType.h"
+#include "velox/type/Date.h"
+
+using facebook::velox::Date;
+using facebook::velox::Timestamp;
+using facebook::velox::TimestampWithTimezone;
+using facebook::velox::core::QueryConfig;
+namespace {
+inline constexpr int64_t kSecondsInDay = 86'400;
+
+FOLLY_ALWAYS_INLINE const date::time_zone* getTimeZoneFromConfig(
```

Comment:
Looks like there is a good amount of code copy-pasted from velox/functions/prestosql/DateTimeFunctions.h . Would it make sense to extract common logic into a header file under velox/functions/lib and re-use in 2 places?

Reply:
fixed. I extracted common logic into DateTimeFormatter.h.